tests/kernel/kselftests.pm

```diff
@@ -0,0 +1,70 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Executes kselftests
+# Maintainer: Kernel QE <kernel-qa@suse.de>
+
+use base 'opensusebasetest';
+use testapi qw(get_var get_required_var set_var);
+use utils;
+use strict;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use LTP::WhiteList;
+
+sub run
+{
+    my ($self) = @_;
+    my $repo = get_var('LINUX_REPO', 'https://github.com/torvalds/linux');
+    my $branch = get_var('LINUX_BRANCH', 'master');
+
+    select_serial_terminal;
+
+    # download linux source code
+    zypper_call("in -y git");
+    assert_script_run("git clone -q --single-branch -b $branch --depth 1 $repo");
+
+    # install build tools and compile tests
+    my $suite = get_required_var('KSELFTESTS_SUITE');
+    my $root = "/root/linux/tools/testing/selftests";
+
+    zypper_call("in -t pattern devel_basis");
+    assert_script_run("make -C $root/$suite", timeout => 1800);
```

Comment:
Everything up to this point is likely unusable for the BPF tests at least. We should use the package (when it is available) which doesn't need a generic module, we can just put it in kselftests_bpf_install.pm or whatever. It's likely that a bunch of quirks will need to be handled during install.

Reply:
Yes, now code is cgroups kselftests specific.

Reply:
Also, the idea is that once we have a `zypper in` we can get rid of all modules for specific kselftests installations and to use a unique variables telling what's the suite we want to install.